firecrown/cosmosis/run.py

```diff
@@ -207,16 +208,117 @@ def _make_cosmosis_values(params):
     cosmosis_values: Inifile
         object to use to build cosmosis parameter ranges/values
     """
+    params = config['parameters']
+    varied_params = config['cosmosis']['parameters']
 
     # copy all the parameters into the cosmosis config structure
     values = {}
+
+    # First set all the numeric parameters, fixed and varied.
+    # We will override the varied ones in a moment
     for p, v in params.items():
-        key = ('params', p)
-        if isinstance(v, list) and not isinstance(v, str):
-            values[key] = ' '.join(str(x) for x in v)
-        else:
-            values[key] = str(v)
+        if isinstance(v, numbers.Number):
+            values['params', p] = str(v)
 
-    cosmosis_values = Inifile(None, override=values)
+    # Now override the varied parameters
+    for p, v in varied_params.items():
+        v = ' '.join(str(x) for x in v)
+        values['params', p] = v
 
-    return cosmosis_values
+    return cosmosis.Inifile(None, override=values)
+
+
+def _make_cosmosis_priors(config):
```

Comment:
Why do we have to redo them for cosmosis. Can't we just use the total output of the priors module like we use for the likelihoods? Also, firecrown will output likelihood+prior. Is there a correction for that I am missing?

Reply:
Because there are other things a priors needs to do other than being evaluated, for many samplers - we also need to be able to sample from it and interpolate into a CDF.  Just having them as a single function like now won't do this.

I will look at supporting the generic scipy.stats distributions in cosmosis.

There is a correction below (might be you saw an older version).